### ISSUE TITLE
HTTPClient connecting to closed port on web export

### ISSUE THREAD

**Issue by EpsylonPT** [godotengine/godot#57124]
### Godot version

3.4.2.stable

### System information

Windows 10

### Issue description

HttpClient's connect_to_host after polling sets its state to HTTPClient.STATUS_CONNECTED in web export,
even though the port is closed.

While running localy the behaviour is the currect one. When exporting to web it is not.

### Steps to reproduce

Run the minimal reproduction project localy, the terminal prints HTTPClient.STATUS_CONNECTING - Correct
Run the minimal reproduction project via godot's Run HTML button, the terminal prints HTTPClient.STATUS_CONNECTED - Incorrect

### Minimal reproduction project

[test.zip](https://github.com/godotengine/godot/files/7926796/test.zip)

**Comment by Calinou** [comment]
cc @Faless

**Comment by Faless** [comment]
No, it's not related, the web export does not use StreamPeerTCP at all.

If anything, this is related to the fetch PR, but I don't have time to bisect now (I think this has always been the case with the web export).

This has to do with the fact that we don't have that information with `fetch`. We could probably stay in connected state instead of switching directly to connected, but might have side effects somewhere else.

**Comment by Faless** [comment]
Actually, I was in a hurry, and I can't double check right now, but IIRC
the reason is that we can't do any connection on the web before requesting,
so we must default to the connected state until the first request is made.
This is, if anything, a documentation issue. So it should be mentioned (if
it's not already) in the exporting for the web section.

**Comment by EpsylonPT** [comment]
@Faless The problem with this is that the make_request always returns OK even when when the server is disconnected. Resulting in a net::ERR_CONNECTION_REFUSED error and TypeError: Failed to fetch in the browser.

Having said this, how could we prevent the request to give a false positive?

**Comment by Faless** [comment]
@EpsylonPT for the way Web API are designed, you cannot avoid such a request from failing and you cannot reliably know if the request is failing in a synchronous way.

So, in our implementation, the HTTPClient status will transition to `STATUS_CONNECTION_ERROR` after some (non-blocking) `poll`.
